### ISSUE TITLE
Idle connections piling up

### ISSUE THREAD

**Issue by tausif-0220** [lcd1232/django-postgrespool2#18]
Hi,
I am facing the issue of too many db idle connections. Earlier I was using psycopg2 engine directly without any pooling. I played with 'CONN_MAX_AGE' parameter and tried to close connection in the application code wherever I felt could help but nothing helped in reality.
I have recently changed to django-postgrespool2 in order to fix the db idle connections. However, I still see the connections are piling up. I have
```
python 3.7.7
django 3.0.4
gunicorn 19.4.5
```

I have the configuration in settings.py as
```
DATABASE_POOL_CLASS = 'sqlalchemy.pool.QueuePool'

DATABASE_POOL_ARGS = {
    'max_overflow': 10,
    'pool_size': 20,
    'recycle': 60
}
```

What is I am doing wrong or missing?

**Comment by lcd1232** [comment]
Hello.

How many are connections idle?

**Comment by tausif-0220** [comment]
Earlier it uses to go as high as in 500. After some changes in the morning today (mostly dB configurations) and using django_postgresspool2 it is in 60-80 range. Reasons for this may be,
I am using gunicorn with 5 workers. I have two different ec2 instances. So in total 5*2*(pool_size + max_overflow) max connections. Still, I think idle connections are high?
I have the following config
`DATABASE_POOL_CLASS = 'sqlalchemy.pool.QueuePool'
DATABASE_POOL_ARGS = {
    'max_overflow': 15,
    'pool_size': 5,
    'recycle': 300,
    'timeout': 60
}`

Also, I am getting this error 

> sqlalchemy.exc.TimeoutError: QueuePool limit of size 5 overflow 15 reached, connection timed out, timeout 60
Is it because the execution unit is not getting enough db connections?

**Comment by tausif-0220** [comment]
UPDATE: I found that at places in our codebase we have been using threading, which was creating # of db connections as the number of threads and django is not closing it. Since django will only close the request thread it serving. All the connections created from the internal threads were remaining in the idle state. As of now, I have removed threading (I tried closing the connection for each thread but the behaviour was intermittent as in not all spawned connections were closed)
Thanks for help.
